apps/bestofjs-nextjs/src/server/api-tags.tsx

```diff
@@ -97,6 +97,34 @@ export function createTagsAPI({ getData }: APIContext) {
         total,
       };
     },
+
+    async getTagBySlug(slug: string) {
+      const { populate, projectCollection, tagCollection } = await getData();
+      const query = new mingo.Query({ code: slug });
+
+      const tags = query
+        .find(tagCollection)
+        .all() as BestOfJS.TagWithProjects[];
+
+      for await (const tag of tags) {
+        const searchQuery = normalizeSearchQuery({
+          criteria: { tags: { $in: [tag.code] } },
+          sort: { stars: -1 },
+          limit: 5,
+          projection: { name: 1, owner_id: 1, icon: 1 },
+        });
+
+        const { projects } = await findRawProjects(
+          projectCollection,
+          searchQuery
+        );
+
+        tag.projects = projects.map(populate);
+      }
+      return {
+        tags,
+      };
+    },
```

Comment:
Since the slug is a unique key, I think it will be better if this API method returns a single tag instead of an array of one one tag: `return { tag: }`.

Reply:
Understood, I will look into fixing this.